modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,125 +1,216 @@
-   PWSTR SymbolicLinkListPtr;
-   GUID Guid = {0x378de44c, 0x56ef, 0x11d1, {0xbc, 0x8c, 0x00, 0xa0, 0xc9, 0x14, 0x05, 0xdd}};
-
-   Status = IoGetDeviceInterfaces(
-      &Guid,
-      NULL,
-      0,
-      &SymbolicLinkList);
-
-   ok(NT_SUCCESS(Status),
-         "IoGetDeviceInterfaces failed with status 0x%X\n",
-         (unsigned int)Status);
-   if (!NT_SUCCESS(Status))
-   {
-      return;
-   }
-
-   DPRINT("IoGetDeviceInterfaces results:\n");
-   for (SymbolicLinkListPtr = SymbolicLinkList;
-        SymbolicLinkListPtr[0] != 0 && SymbolicLinkListPtr[1] != 0;
-        SymbolicLinkListPtr += wcslen(SymbolicLinkListPtr) + 1)
-   {
-      DPRINT1("Symbolic Link: %S\n", SymbolicLinkListPtr);
-   }
+    &KSCATEGORY_BRIDGE,
+    &KSCATEGORY_CAPTURE,
+    &KSCATEGORY_COMMUNICATIONSTRANSFORM,
+    &KSCATEGORY_DATACOMPRESSOR,
+    &KSCATEGORY_DATADECOMPRESSOR,
+    &KSCATEGORY_DATATRANSFORM,
+    &KSCATEGORY_FILESYSTEM,
+    &KSCATEGORY_INTERFACETRANSFORM,
+    &KSCATEGORY_MEDIUMTRANSFORM,
+    &KSCATEGORY_MIXER,
+    &KSCATEGORY_RENDER,
+    &KSCATEGORY_SPLITTER,
+};
 
-#if 0
-   DPRINT("[PnP Test] Trying to get aliases\n");
+static
+VOID
+Test_IoOpenDeviceInterfaceRegistryKey(PWSTR SymbolicLink)
+{
+    UNICODE_STRING KeyName, SymbolicLinkName;
 
-   for (SymbolicLinkListPtr = SymbolicLinkList;
-        SymbolicLinkListPtr[0] != 0 && SymbolicLinkListPtr[1] != 0;
-        SymbolicLinkListPtr += wcslen(SymbolicLinkListPtr) + 1)
-   {
-      UNICODE_STRING SymbolicLink;
-      UNICODE_STRING AliasSymbolicLink;
-
-      SymbolicLink.Buffer = SymbolicLinkListPtr;
-      SymbolicLink.Length = SymbolicLink.MaximumLength = wcslen(SymbolicLinkListPtr);
-      RtlInitUnicodeString(&AliasSymbolicLink, NULL);
-      IoGetDeviceInterfaceAlias(
-         &SymbolicLink,
-         &AliasGuid,
-         &AliasSymbolicLink);
-      if (AliasSymbolicLink.Buffer != NULL)
-      {
-         DPRINT("[PnP Test] Original: %S\n", SymbolicLinkListPtr);
-         DPRINT("[PnP Test] Alias: %S\n", AliasSymbolicLink.Buffer);
-      }
-   }
-#endif
-
-   ExFreePool(SymbolicLinkList);
+    RtlInitUnicodeString(&SymbolicLinkName, SymbolicLink);
+    RtlInitUnicodeString(&KeyName, L"ReactOS_kmtest");
+
+    for (size_t n = 0; n < RTL_NUMBER_OF(Types); ++n)
+    {
+        HANDLE DeviceInterfaceKey, DeviceInterfaceSubKey;
+        OBJECT_ATTRIBUTES ObjectAttributes;
+
+        NTSTATUS Status = IoOpenDeviceInterfaceRegistryKey(&SymbolicLinkName, KEY_CREATE_SUB_KEY, &DeviceInterfaceKey);
+
+        if (NT_SUCCESS(Status))
+        {
+            DPRINT1("IoOpenDeviceInterfaceRegistryKey(): success: %d %p\n", n, DeviceInterfaceKey);
+        }
+        else
+        {
+            DPRINT1("IoOpenDeviceInterfaceRegistryKey(): fail: %d 0x%x\n", n, Status);
+        }
+
+        InitializeObjectAttributes(
+            &ObjectAttributes,
+            &KeyName,
+            OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+            DeviceInterfaceKey,
+            NULL);
```

Comment:
```suggestion
        InitializeObjectAttributes(&ObjectAttributes,
                                   &KeyName,
                                   OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
                                   DeviceInterfaceKey,
                                   NULL);
```
Idem for the other call below.

Reply:
Fixed.